src/main/java/com/aws/greengrass/util/BatchedSubscriber.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util;
+
+import com.aws.greengrass.config.ChildChanged;
+import com.aws.greengrass.config.Node;
+import com.aws.greengrass.config.Subscriber;
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.config.WhatHappened;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiPredicate;
+
+/**
+ * Utility class that allows a subscription to only fire
+ * on the last change in a batch.
+ *
+ * <p>This is useful in scenarios such as configuration changes
+ * where you may only want to perform expensive work once within
+ * a short span.
+ */
+public final class BatchedSubscriber {
+
+    private final Runnable unsubscribe;
+
+    private BatchedSubscriber(Runnable unsubscribe) {
+        this.unsubscribe = unsubscribe;
+    }
+
+    /**
+     * Remove the subscription.
+     */
+    public void unsubscribe() {
+        if (unsubscribe != null) {
+            unsubscribe.run();
+        }
+    }
+
+    /**
+     * Subscribe to topic changes.
+     *
+     * @param topic      topic
+     * @param afterBatch callback to perform after a batch of changes
+     * @return batched subscriber
+     */
+    public static BatchedSubscriber subscribe(Topic topic, Runnable afterBatch) {
+        return subscribe(topic, afterBatch, null, null);
+    }
+
+    /**
+     * Subscribe to topic changes. Subscriptions will only be called once
+     * for a batch of changes.
+     *
+     * @param topic            topic
+     * @param afterBatch       callback to perform after a batch of changes
+     * @param onInitialization callback to perform after the subscription has been added
+     * @param excludeFilter    custom filter for ignoring changes
+     * @return batched subscriber
+     */
+    public static BatchedSubscriber subscribe(Topic topic,
+                                              Runnable afterBatch, Runnable onInitialization,
+                                              BiPredicate<WhatHappened, Topic> excludeFilter) {
+        AtomicInteger numRequestedChanges = new AtomicInteger();
+        AtomicInteger totalCalls = new AtomicInteger();
+        Subscriber subscriber = (what, child) -> {
```

Comment:
instead of creating this subscriber and atomic integers inline, store them as fields of this class. Have BatchedSubscriber implement the ChildChanged interface so you can then just do `topic.subscribe(this)`, `topic.remove(this)`